gdpr/impl.go

```diff
@@ -98,11 +111,13 @@ func (p *permissionsImpl) allowSync(ctx context.Context, vendorID uint16, consen
 	return false, nil
 }
 
-func (p *permissionsImpl) allowPI(ctx context.Context, vendorID uint16, consent string) (bool, bool, bool, error) {
-	// If we're not given a consent string, respect the preferences in the app config.
-	if consent == "" {
+func (p *permissionsImpl) allowPI(ctx context.Context, vendorID uint16, gdpr Signal, consent string) (bool, bool, bool, error) {
+	if consent == "" && gdpr == AmbiguousGDPR {
 		return p.cfg.UsersyncIfAmbiguous, p.cfg.UsersyncIfAmbiguous, p.cfg.UsersyncIfAmbiguous, nil
 	}
```

Comment:
If `allowPI` is only ever called from `PersonalInfoAllowed`, consider moving up the `consent == "" && gdpr == AmbiguousGDPR` check before the vendor check, since it's done first by either possible branch.

Reply:
I confirmed this is only called from `PersonalInfoAllowed` so I moved this and the `if consent == "" && gdprSignal == SignalYes {` check immediately after it before the vendor check.